src/main/java/org/kuali/kfs/module/purap/document/web/struts/PurchasingActionBase.java

```diff
@@ -1647,9 +1655,11 @@ private void createReasonNote(ActionForm form) {
     	KualiDocumentFormBase kualiDocumentFormBase = (KualiDocumentFormBase) form;
         try {
             Note noteObj = SpringContext.getBean(DocumentService.class).createNoteFromDocument(kualiDocumentFormBase.getDocument(), ((PurchasingFormBase)kualiDocumentFormBase).getReasonToChange());
+            populateIdentifierOnNoteAndExtension(noteObj);
             kualiDocumentFormBase.getDocument().addNote(noteObj);
-         //   SpringContext.getBean(NoteService.class).save(noteObj);
-            SpringContext.getBean(NoteService.class).saveNoteList(kualiDocumentFormBase.getDocument().getNotes());
+            if (isNoteReadyToBeSaved(kualiDocumentFormBase.getDocument(), noteObj)) {

```

Comment:
There are other types of documents besides REQS that could be using this functionality, and as per the "getPersistedBoNotesNotes" method, the notes being checked for requiring reason-note generation are not necessarily from the current document (if it's not a REQS). If there are other documents besides REQS that can alter the "Send to Vendor" flag while still in INITIATED status, then I think we need some form of the "isNoteReadyToBeSaved" logic. Otherwise, we can remove this extra check.

As per Sandy's suggestion, I will also add a second reviewer to this PR to get feedback on these code changes and this question.